### PR TITLE
[CINN] fix concat

### PR DESCRIPTION
**Pull request by wangzhen38**
### PR types
Others

### PR changes
Others

### Describe
更正concat算子拆解的注册位置：将反向maker写在前向注册里

### CODE REVIEW

File: paddle/fluid/operators/concat_op.cc [hunk 1]
```diff
@@ -165,7 +165,7 @@ class ConcatCompositeGradOpMaker : public prim::CompositeGradOpMakerBase {
     paddle::optional<paddle::Tensor> tensor_axis =
         this->GetOptionalSingleForwardInput("AxisTensor");
     paddle::Tensor out_grad = this->GetSingleOutputGrad("Out");
-    std::vector<paddle::Tensor> input_grad = this->GetMultiForwardInput("X");
+    std::vector<paddle::Tensor> input_grad = this->GetMultiInputGrad("X");
```
Comment (JiabinYang) [inline paddle/fluid/operators/concat_op.cc:168]: change to `std::vector<paddle::Tensor> input_grad = this->GetMultiInputGrad("X", false);`

**Review by JiabinYang — COMMENTED** [review]
one comments

File: paddle/fluid/operators/concat_op.cc (hunk 1 above)
Comment (wangzhen38) [reply paddle/fluid/operators/concat_op.cc:168]: done

**Review by Charles-hit — APPROVED** [review]

File: python/paddle/fluid/tests/unittests/test_concat_op.py [hunk 2]
```diff
@@ -266,6 +279,47 @@ def setUp(self):
 
 def create_test_fp16(parent):
     class TestConcatFp16(parent):
+        def setUp(self):
+            self.op_type = "concat"
+            self.python_api = paddle.concat
+            self.public_python_api = paddle.concat
+            self.enable_cinn = False
```
Comment (xiaoguoguo626807) [inline python/paddle/fluid/tests/unittests/test_concat_op.py:286]: if don't check prim no need to add enable_cinn

**Review by xiaoguoguo626807 — APPROVED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/test_concat_op.py (hunk 2 above)
Comment (wangzhen38) [reply python/paddle/fluid/tests/unittests/test_concat_op.py:286]: 稍后会提pr修改

**Review by zyfncg — APPROVED** [review]
LGTM for op_compat.yaml

**Pull request merged**